src/configureWorkspace/configureDotNetCore.ts

```diff
@@ -267,20 +257,101 @@ function genDockerFile(serviceNameAndRelativePath: string, platform: Platform, o
     const scaffoldContext: DockerDebugScaffoldContext = {
         folder: folder,
         platform: 'netCore',
         actionContext: context,
-        dockerfile: dockerfile,
+        // always use posix for debug config because it's committed to source control and works on all OS's
+        // tslint:disable-next-line: no-invalid-template-strings
+        dockerfile: path.posix.join('${workspaceFolder}', workspaceRelativeDockerfileName),
     }
 
     const options: NetCoreScaffoldingOptions = {
-        // always use posix for debug config because it's committed to source control and works on all OS's
         // tslint:disable-next-line: no-invalid-template-strings
-        appProject: path.posix.join('${workspaceFolder}', artifactName),
+        appProject: path.posix.join('${workspaceFolder}', workspaceRelativeProjectFileName),
         platformOS: platformOS,
     }
 
     await dockerDebugScaffoldingProvider.initializeNetCoreForDebugging(scaffoldContext, options);
 }
+
+// tslint:disable-next-line: export-name
+export async function scaffoldNetCore(context: ScaffolderContext): Promise<ScaffoldFile[]> {
+    const os = context.os ?? await context.promptForOS();
+
+    const telemetryProperties = <ConfigureTelemetryProperties>context.telemetry.properties;
+
+    telemetryProperties.configureOs = os;
+
+    const ports = context.ports ?? (context.platform === 'ASP.NET Core' ? await context.promptForPorts([80, 443]) : undefined);
+
+    const rootRelativeProjectFileName = await context.captureStep('project', findCSProjOrFSProjFile)(context.rootFolder);
+    const rootRelativeProjectDirectory = path.dirname(rootRelativeProjectFileName);
+
+    telemetryProperties.packageFileType = path.extname(rootRelativeProjectFileName);
+    telemetryProperties.packageFileSubfolderDepth = getSubfolderDepth(context.rootFolder, rootRelativeProjectFileName);
+
+    const projectFilePath = path.posix.join(context.rootFolder, rootRelativeProjectFileName);
+    const workspaceRelativeProjectFileName = path.posix.relative(context.folder.uri.fsPath, projectFilePath);
+
+    let serviceNameAndPathRelative = rootRelativeProjectFileName.slice(0, -(path.extname(rootRelativeProjectFileName).length));
+    const projFileContents = (await fse.readFile(path.join(context.rootFolder, rootRelativeProjectFileName))).toString();
+
+    // Extract TargetFramework for version
+    const [version] = extractRegExGroups(projFileContents, /<TargetFramework>(.+)<\/TargetFramework/, ['']);
+
+    if (context.outputFolder) {
+        // We need paths in the Dockerfile to be relative to the output folder, not the root
+        serviceNameAndPathRelative = path.relative(context.outputFolder, path.join(context.rootFolder, serviceNameAndPathRelative));
+    }
+
+    serviceNameAndPathRelative = serviceNameAndPathRelative.replace(/\\/g, '/');
```

Comment:
Why is this necessary? Can you add a comment?

Reply:
Is there a standard library API that could be used for this?

Reply:
It's to ensure that the path (which may come from a Windows filesystem) uses POSIX separators in the resulting Dockerfile (which most likely targets Linux but, even if Windows, will still be valid).  You'd think that `path.posix.normalize()` would do that but, sadly, it doesn't convert Windows separators to POSIX separators.  I've added a comment.